File: Task 1/Java/app/src/main/java/dev/ogabek/java/MyApplication.java
package dev.ogabek.java;

import android.app.Application;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class MyApplication extends Application {

    public static MyApplication instance = null;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    RequestQueue requestQueue = null;

    private RequestQueue getRequestQueue() {
        if (requestQueue == null) {
            return Volley.newRequestQueue(getApplicationContext());
        }
        else  {
            return requestQueue;
        }
    }

    public <T> void addToRequestQueue(Request<T> request) {
        getRequestQueue().add(request);
    }

}

File: Task 2/Kotlin/app/src/main/java/dev/ogabek/kotlin/MyApplication.kt
package dev.ogabek.kotlin

import android.app.Application
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    private val requestQueue: RequestQueue? = null
        get() {
            return field ?: Volley.newRequestQueue(applicationContext)
    }

    fun <T> addToRequestQueue(request: Request<T>?) {
        requestQueue!!.add(request)
    }

    companion object {
        var instance: MyApplication? = null
    }
}